src/renderer/components/table-list/table-list.tsx

```diff
@@ -0,0 +1,167 @@
+import type { SortingState, Table as TableType, ColumnDef } from "@tanstack/react-table";
+import { useReactTable, getCoreRowModel, getSortedRowModel } from "@tanstack/react-table";
+import React, { HTMLProps, useMemo } from "react";
+import { createColumnHelper } from "@tanstack/react-table";
+import { Icon } from "../icon";
+import { Menu, MenuItem } from "../menu";
+import { withInjectables } from "@ogre-tools/injectable-react";
+import getRandomIdInjectable from "../../../common/utils/get-random-id.injectable";
+import { VirtualTable } from "../table/virtual-table";
+
+interface TableProps<T> {
+  columns: ColumnDef<T, any>[];
+  data: T[];
+  className?: string;
+  selectable?: boolean;
+  configurable?: boolean;
+  columnsResizable?: boolean;
+  onRowClick?: (item: T) => void;
+}
+
+export function TableList<Data>({
+  columns,
+  data,
+  className,
+  selectable = true,
+  configurable = true,
+  columnsResizable = true,
+  onRowClick,
```

Comment:
Too many responsibilities for something called "TableList". Shouldn't UI code for something like "TableList" only care what it looks like, and let other code take care of other responsibilities?